src/encoded/static/components/filegallery.js

```diff
@@ -1822,29 +2090,74 @@ class FileGalleryRendererComponent extends React.Component {
+        this.getAvailableBrowsers = this.getAvailableBrowsers.bind(this);
+        this.resetCurrentBrowser = this.resetCurrentBrowser.bind(this);
     }
 
-    // Set the default filter after the graph has been analyzed once.
     componentDidMount() {
+        // Set the default filter after the graph has been analyzed once.
         if (!this.props.altFilterDefault) {
             this.setState({ selectedFilterValue: '0' });
         }
+        // Determine how many visualizable files there are
+        // const tempFiles = this.state.files.filter(file => (
+        //     (file.file_format === 'bigWig' || file.file_format === 'bigBed')
+        //     && (file.file_format_type !== 'bedMethyl')
+        //     && (file.file_format_type !== 'bedLogR')
+        //     && (file.file_format_type !== 'idr_peak')
+        //     && (file.file_format_type !== 'tss_peak')
+        //     && (file.file_format_type !== 'pepMap')
+        //     && (file.file_format_type !== 'modPepMap')
+        //     && ['released', 'in progress', 'archived'].indexOf(file.status) > -1
+        // ));
+        // If the graph is hidden and there are no visualizable files, set default tab to be table and set default assembly to be 'All assemblies'
+        // if (this.props.hideGraph && tempFiles.length < 1) {
+        if (this.props.hideGraph) {
+            this.setState({ currentTab: 'tables' }, () => {
+                this.filterFiles('All assemblies', 'assembly');
+            });
+        // If the graph is not hidden and there are no visualizable files, set default tab to be graph and set default assembly to be the most recent assembly
+        } else {
+        // } else if (tempFiles.length < 1) {
+            // Display graph as default if there are no visualizable files
+            let assemblyList = [];
+            this.setState({ currentTab: 'graph' }, () => {
+                // Determine available assemblies
+                assemblyList = this.setAssemblyList(this.state.files);
+                // We want to get the assembly with the highest assembly number (but not 'All assemblies')
+                const newAssembly = Object.keys(assemblyList).reduce((a, b) => (((assemblyList[a] > assemblyList[b]) && (a !== 'All assemblies')) ? a : b));
+                this.filterFiles(newAssembly, 'assembly');
+            });
+        // If there are visualizable files, set default tab to be browser and set default assembly to be the most recent assembly
+        }
+        //  else {
+        //     // Determine available assemblies
+        //     const assemblyList = this.setAssemblyList(this.state.files);
+        //     // We want to get the assembly with the highest assembly number (but not 'All assemblies')
+        //     const newAssembly = Object.keys(assemblyList).reduce((a, b) => (((assemblyList[a] > assemblyList[b]) && (a !== 'All assemblies')) ? a : b));
+        //     this.filterFiles(newAssembly, 'assembly');
+        // }
```

Comment:
I know it is bad form to leave commented code in production but I was considering leaving this because I think eventually we will probably want the default tab to the the browser tab and the commented code here sets the default to be the browser if there are visualizable files.

Reply:
I think Emma knows this but in case others don’t, webpack strips all comments from production builds, so the code that actually gets run on demos and production won’t see any of this anyway — it won’t even take up space.